go/multipooler/manager/pg_multischema.go

```
@@ -0,0 +1,368 @@
+	pm.logger.InfoContext(ctx, "Creating multigres sidecar schema")
+
+	tableGroup := pm.getTableGroup()
+
+	// MVP validation: only default tablegroup is supported
+	if tableGroup != defaultTableGroup {
+		pm.logger.ErrorContext(ctx, "Only default tablegroup is supported in MVP",
+			"tablegroup", tableGroup)
+		return mterrors.New(mtrpcpb.Code_FAILED_PRECONDITION,
+			"only default tablegroup is supported, got: "+tableGroup)
+	}
+
+	if err := pm.createSchema(ctx); err != nil {
+		return err
+	}
+
+	if err := pm.createHeartbeatTable(ctx); err != nil {
+		return err
+	}
+
+	if err := pm.createDurabilityPolicyTable(ctx); err != nil {
+		return err
+	}
+
+	// Create multischema global tables for the default tablegroup
+	pm.logger.InfoContext(ctx, "Creating multischema global tables for default tablegroup")
+
+	if err := pm.createTablegroups(ctx); err != nil {
+		return err
+	}
+
+	if err := pm.createTables(ctx); err != nil {
+		return err
+	}
+
+	if err := pm.createShards(ctx); err != nil {
+		return err
+	}
+
+	pm.logger.InfoContext(ctx, "Successfully created multigres sidecar schema")
+	return nil
+}
+
+// initializeMultischemaData inserts the initial tablegroup and shard records.
+//
+// MVP Limitation: Currently, we only support the default tablegroup with shard "0-inf".
+// This function validates these constraints and returns an error otherwise.
+//
+// TODO: In the future, tablegroup and shard insertion should be done via a dedicated
+// RPC, and the bootstrap code should insert the tablegroup in the default primary
+// pooler. For simplicity in the MVP, we do this as part of InitializePrimary since
+// we only support a single tablegroup/shard for now.
+func (pm *MultiPoolerManager) initializeMultischemaData(ctx context.Context) error {
+	tableGroup := pm.getTableGroup()
+	shard := pm.getShard()
+
+	// MVP validation: only default tablegroup with shard 0-inf is supported
+	if tableGroup != defaultTableGroup {
```

to fail faster, would it make sense to move or duplicate this validation logic in `MultiPooler.Init()`?